src/util.rs

```diff
@@ -419,3 +419,165 @@ fn get_network_interfaces_impl() -> Vec<NetworkInterface> {
 
     vec
 }
+
```

Comment:
I notice there are no tests with use `Some(_)` for the second parameter - could some be added? I know it's done as part of the other tests, I'd like there to be some checksum specific test cases though for completeness.